modules/matrix.lobster

```diff
@@ -0,0 +1,42 @@
+import vec
+
+// A 2x2 matrix representated as a 4 length vector of x,y,z,w so that
+// |x y|
+// |z w|
+
+struct mat2x2 : xyzw_f
+
+    def operator*(o:mat2x2):
+        // This could be expressed with 4 dot product if `this.zw` existed (and 
+        // others)
+        return mat2x2 { x*o.x + y*o.z, x*o.y + y*o.w,
+                        z*o.x + w*o.z, z*o.y + w*o.w }  
+    
+    def operator*(o:xy_f):
+        // This would be nicer if `this.zw` existed.
```

Comment:
just add `zw` in `vec.lobster`

Reply:
I was worried that it could be confusing, since we are shifting the z -> x and w-> y, but if you are OK with it.

Reply:
that's the same as in GLSL, if you do `.zw` you'll get an xy vector back